core/src/main/java/org/elasticsearch/index/query/WrapperQueryBuilder.java

```diff
@@ -162,8 +162,8 @@ protected boolean doEquals(WrapperQueryBuilder other) {
     @Override
     protected QueryBuilder<?> doRewrite(QueryRewriteContext context) throws IOException {
         try (XContentParser qSourceParser = XContentFactory.xContent(source).createParser(source)) {
-            QueryParseContext parseContext = context.newParseContext();
-            parseContext.reset(qSourceParser);
+            qSourceParser.setParseFieldMatcher(context.indexSettings.getParseFieldMatcher());
```

Comment:
good point, I think thats a good idea. Only one is called parseFieldMatcher, the other getParseFieldMatcher, I guess we are leaning to getParseFieldMatcher then.